xstate-chatbot/nodejs/src/machine/vitals.js

```diff
@@ -55,6 +98,106 @@ const vitalsFlow = {
           }),
           always: 'prompt'
         }
+     }
+    },
+    rrtMobileNumber: {
+      id: 'rrtMobileNumber',
+      initial: 'prompt',
+      states: {
+        prompt: {
+          onEntry: assign((context, event) => {
+            dialog.sendMessage(context, dialog.get_message(messages.rrtMobileNumber.prompt, context.user.locale));
+          }),
+          on: {
+            USER_MESSAGE: 'process'
+          }
+        },
+        process: {
+          onEntry: assign((context, event) => {
+            context.message = dialog.get_input(event, false);
+          }),
+          invoke: {
+            src: (context, event) => vitalsService.getPatientDetailsFromMobileNumber(context.message),
+            onDone: [
+              {
+                cond: (context, event) => event.data.data === null,
+                actions: assign((context, event) => {
+                  context.persons = event.data;
```

Comment:
Any need for this? It will anyways be null. Remove the action if is not needed.

Reply:
null is there when no record is found fro any mobile number. like

async getPatientDetailsFromMobileNumber(mobileNumber) {return {"response": 0,"sys_message": "No Record Found.","data": null}};

Reply:
Still, is there any need to save the event.data value in context when data.data is null?

Reply:
removed